iscsi/get.go

```diff
@@ -0,0 +1,314 @@
+// Copyright 2017 The Prometheus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package iscsi
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+var sysPath = ""
+
+// GetStats is the main iscsi status information func
+// building the path and prepare info for enable iscsi
+func GetStats(rootPath string, iqnPath string) (*Stats, error) {
+	var istats Stats
+
+	istats.Name = filepath.Base(iqnPath)
+	istats.RootPath = rootPath
+	sysPath = istats.RootPath
+	matches, err := filepath.Glob(filepath.Join(iqnPath, "tpgt*"))
+	if err != nil {
+		return nil, fmt.Errorf("lio: get TPGT error %v", err)
+	}
+	istats.Tpgt = make([]TPGT, len(matches))
+
+	for ii, tpgtPath := range matches {
+		istats.Tpgt[ii].Name = filepath.Base(tpgtPath)
+		istats.Tpgt[ii].TpgtPath = tpgtPath
+		istats.Tpgt[ii].IsEnable, _ = isPathEnable(tpgtPath)
+		if istats.Tpgt[ii].IsEnable {
+			matchesLunsPath, _ := getLun(tpgtPath)
+			istats.Tpgt[ii].Luns = make([]LUN, len(matchesLunsPath))
+
+			for ll, lunPath := range matchesLunsPath {
+				backstore, objectName, typeNumber, err := getLunLinkTarget(lunPath)
+				if err != nil {
+					continue
+				}
+				istats.Tpgt[ii].Luns[ll].Name = filepath.Base(lunPath)
+				istats.Tpgt[ii].Luns[ll].LunPath = lunPath
+				istats.Tpgt[ii].Luns[ll].Backstore = backstore
+				istats.Tpgt[ii].Luns[ll].ObjectName = objectName
+				istats.Tpgt[ii].Luns[ll].TypeNumber = typeNumber
+			}
+		}
+	}
+	return &istats, nil
+}
+
+// isPathEnable is a utility function
+// check if the file "enable" contain enable message
+func isPathEnable(path string) (bool, error) {
+	var isEnable bool
+	isEnable = false
+	tmp, err := ioutil.ReadFile(filepath.Join(path, "enable"))
```

Comment:
Should use a more meaningful variable name here instead of `tmp`

Reply:
done, enableReadout it is